### ISSUE TITLE
Add relaxed option for no-relative-parent-imports

### ISSUE THREAD

**Issue by ninjaprox** [benmosher/eslint-plugin-import#1243]
This is the proposal for `no-relative-parent-imports` rule improvement.

### Context

When `no-relative-parent-imports` rule is enabled, it doesn't allow to import from relative parents at all, for example:

```javascript
// Both will fail ESLint
import component from '../path/to/component' // immediate parent
import component from '../../path/to/component'
```

However, it's sometimes acceptable and reasonable to import from the immediate parent as depicted above.

### Proposal

Give `no-relative-parent-imports` the option of either `strict` or `relaxed`.

- If `strict` is used, nothing will change, the current behavior stays intact. `strict` is by default if not specifying the option.

```javascript
// Both will be in strict mode
// Explicitly specify the option
{
  `no-relative-parent-imports`: ['warn', 'strict']
}
// `strict` by default
{
  `no-relative-parent-imports`: 'warn'
}
```

- If `relaxed` is used, the rule will pass on immediate parent imports, and still catch the rest.

```javascript
{
  `no-relative-parent-imports`: ['warn', 'relaxed']
}
```

```javascript
import component from '../path/to/component' // passed
import component from '../../path/to/component' // failed
```

### Effect

By using `strict` by default, this proposal doesn't affect any existing users. However, it gives users more options to tweak the rule if needed.

### Alternatives

Give the option to specify how depth of relative parent imports is allowed.

```javascript
{
  `no-relative-parent-imports`: ['warn', { depth: 2 }]
}
```

```javascript
import component from '../path/to/component' // passed
import component from '../../path/to/component' // passed
import component from '../../../path/to/component' // passed
```

However, I think this isn't a preferred option since it goes against the purpose of `no-relative-parent-imports` rule.

**Comment by ljharb** [comment]
Indeed; I think that rule you actually want is a different one - more like enforcing import boundaries on specific directories.

**Comment by ninjaprox** [comment]
That might become a rule as well, and that will give users fine-grained control over how to enforce imports. This proposal is far simpler than that rule in term of configuration and enforcement, yet opens flexibility.

**Comment by sharikovvladislav** [comment]
Any information on this?

**Comment by graphan** [comment]
Bumping up.
